core/src/test/scala/za/co/absa/spline/core/SparkLineageInitializerSpec.scala

```diff
@@ -47,25 +49,35 @@ object SparkLineageInitializerSpec {
     override val createDataLineageReader: Option[DataLineageReader] = sys.error("bam!")
   }
 
-  private[this] def sparkQueryExecutionListenerClasses: Seq[Class[_ <: QueryExecutionListener]] = {
-    val session = SparkSession.builder.getOrCreate
+  private[this] def sparkQueryExecutionListenerClasses(session: SparkSession): Seq[Class[_ <: QueryExecutionListener]] = {
     (session.listenerManager.getClass.getDeclaredFields collectFirst {
       case f if f.getName endsWith "listeners" =>
         f setAccessible true
         (f get session.listenerManager).asInstanceOf[Seq[QueryExecutionListener]].map(_.getClass)
     }).get
   }
 
-  private def assertSplineIsEnabled() = sparkQueryExecutionListenerClasses should contain(classOf[SplineQueryExecutionListener])
+  private def assertSplineIsEnabled(session: SparkSession = SparkSession.builder().getOrCreate()) =
+    sparkQueryExecutionListenerClasses(session) should contain(classOf[SplineQueryExecutionListener])
+
+  private def assertSplineIsDisabled(session: SparkSession = SparkSession.builder().getOrCreate()) =
+    sparkQueryExecutionListenerClasses(session) should not contain classOf[SplineQueryExecutionListener]
```

Comment:
nit: inconsistent braces around `classOf`

Reply:
also previous code issue.

Reply:
Interesting that the `not` operator causes that if you try to add brackets for the second contain the IDEA will raise warning that the brackets are not necessary. The first brackets cannot be removed. So I am keeping as is.